Source/VMware.vSphereDSC/Tests/Unit/VMHostVss.Unit.Tests.ps1

```diff
@@ -0,0 +1,716 @@
+                NumPorts = $($script:resourceProperties.NumPorts)
+            }
+        }
+'@
+    $script:updateNetworkCode = @'
+        return [VMware.Vim.HostNetworkConfigResult] @{
+            ConsoleVnicDevice = @()
+            VnicDevice = @()
+        }
+'@
+
+    $env:PSModulePath = $script:mockModuleLocation
+    $vimAutomationModule = Get-Module -Name VMware.VimAutomation.Core
+    if ($null -ne $vimAutomationModule -and $vimAutomationModule.Path -NotMatch 'TestHelpers') {
+        throw 'The Original VMware.VimAutomation.Core Module is loaded in the current session. If you want to run the unit tests please open a new PowerShell session.'
+    }
+
+    Import-Module -Name VMware.VimAutomation.Core
+
+    $script:viServer = [VMware.Vim.VIServer] @{
+        Name = $script:resourceProperties.Server
+        User = $script:Constants.VIServerUser
+    }
+    $script:networkSystemMoRef = [VMware.Vim.ManagedObjectReference] @{
+        Type = $script:Constants.NetworkSystemMoRefType
+        Value = $script:Constants.NetworkSystemMoRefValue
+    }
+}
+
+function AfterAllTests {
+    Remove-Module -Name VMware.VimAutomation.Core
+    $env:PSModulePath = $script:modulePath
+}
+
+# Calls the function to Import the mocked VMware.VimAutomation.Core module before all tests.
+Try {
+    BeforeAllTests
+
+    Describe 'VMHostVss\Set'  -Tag 'Set' {
+        Context 'Present + VSS does not exist' {
+            BeforeAll {
+                $viserverMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:viServerCode))
+                $vmHostMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:vmHostCode))
+                $networkSystemMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:networkSystemNoVssCode))
+                $vssConfigMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:vssConfigAddVssCode))
+                $updateNetworkMock = [scriptblock]::Create($ExecutionContext.InvokeCommand.ExpandString($script:updateNetworkCode))
+            }
+
+            # Arrange
+            Mock -CommandName Connect-VIServer -MockWith $viServerMock -ModuleName $script:moduleName
+            Mock -CommandName Get-VMHost -MockWith $vmHostMock -ModuleName $script:moduleName
+            Mock -CommandName Get-View -MockWith $networkSystemMock -ModuleName $script:moduleName
+            Mock -CommandName New-VssConfig -MockWith $vssConfigMock -ModuleName $script:moduleName
+            Mock -CommandName Update-Network -MockWith $updateNetworkMock -ModuleName $script:moduleName
+
+            $resource = New-Object -TypeName $script:resourceName -Property $script:resourceProperties
+
+            It 'Should call the Connect-VIServer mock with the passed server and credentials once' {
+                # Act
+                $resource.Set()
+
+                # Assert
+                $sAssert = @{
```

Comment:
Maybe here we should think about better naming.

Reply:
You mean the name of the splat variable?

Reply:
Yes.

Reply:
Any suggestions?
That's my personal convention, 's' for splat variable and then the verb of the cmdlet where I use it.
And it is a local variable for the It block.

Reply:
I was thinking of something like $assertMockCalledParams.

Reply:
Done